tests/installation/change_desktop.pm

```diff
@@ -60,8 +60,8 @@ sub change_desktop {
             send_key ' ';
         }
         if (check_var('DESKTOP', 'textmode')) {
-            send_key_until_needlematch 'x11-selected', 'down';
-            send_key ' ';
+            send_key_until_needlematch [qw(x11-selected x11-unselected)], 'down';
+            send_key ' ' if match_has_tag('x11-selected');
```

Comment:
what's the purpose of that change? Can you crosscheck the scenario https://openqa.suse.de/tests/1305115#step/change_desktop/10 ?

Reply:
Installing SLES4SAP with `VIDEOMODE=text` set, when GNOME Desktop Environment is removed from the packages list, it also removes X Windows System automatically, as can be seen here: http://mango.suse.de/tests/129#step/change_desktop/8

In SLES this is not happening; when GNOME Desktop Environment is removed from the packages list, X Windows System is still marked for installation (as can be seen in the link you pasted).

So the current version of installation/change_desktop in upstream, sends the down key until the x11-selected needle matches and removes X Windows System explicitly (in SLES).

Leaving the code as is, causes the test to fail with SLES4SAP, because X Windows System was removed at the same time as GNOME Desktop Environment, so the x11-selected needle never matches. It fails with a timeout IIRC.

My workaround is to send the down key until it matches either x11-selected and x11-unselected (so, basically, until I have moved the selection bar to the X Windows System package, regardless of whether it is currently selected or not), and then only send the space key to remove X Windows System if the matched needle was x11-selected because this part of the code is within an `if` block for when DESKTOP is textmode.